matrix_factorization/loss.cu

```diff
@@ -9,30 +9,35 @@
 #include "matrix.h"
 
 #define index(i, j, N)  ((i)*(N)) + (j)
+#define warp_size 32 //TODO: we need to get device props
 
 using namespace cu2rec;
 
 // PARALLEL
+extern __shared__ float biases[];
```

Comment:
Does this need to be outside? Can probably go inside the `loss_kernel`.

Reply:
it did when there were two shared mem arrays, now there are one. So yeah, we can put it back inside